### ISSUE TITLE
Support optimized path for raw queries

### ISSUE THREAD

**Issue by fako1024** [els0r/goProbe#213]
As previously discussed in #75 we are currently very limited when it comes to performing raw queries, simply because everything is read into a map, then sorted, then filled into a result. Only then the data is sent to the output. Since a raw query technically isn't much more than a "dump" of the data on disk it would be prudent if we treated it as such and basically just stream the data to the output, bypassing all unnecessary steps / conversions.
In theory the goal should be to support raw queries of any time range even on low memory devices.

DoD

- [ ] Support building slices of results in special path of `DBWorkManager` for raw queries
- [ ] Consider creation extension of printer interface that supports streaming chunks of results
- [ ] ...

**Comment by els0r** [comment]
Makes sense. The "sort-by-time" is solved already, since data is prepared to be read from disk temporally ordered.

We could even support conditions on `raw` queries.

The only thing that kept me clinging on to the current approach was the fact that results were nicely formatted with the table output. To be fair, it feels like a weak argument to _not_ pursue other means to read and present the data (a.k.a. sequential printing of the data). Still, we would/should probably ditch the idea of properly formatting the results and just spew them onto stdout with a delimiter (comma, space, or otherwise configured).

I foresee some effort regarding results streaming. We for sure need to extend the data structure to support it meaningfully, as in:

- stream a bulk of rows per response
- stream increments of the summary?

Let's get some insight on how scenarios such as these are usually handled and then go ahead.

**Comment by fako1024** [comment]
> We could even support conditions on `raw` queries.
Definitively a _must_ feature (since we have to implement a secondary processing pipeline in the `DBWorker` anyway there's no reason not to adapt the existing conditional handling to fit the raw scenario), but I don't see an issue with that.

> Still, we would/should probably ditch the idea of properly formatting the results and just spew them onto stdout with a delimiter (comma, space, or otherwise configured).
My initial though process was to have the raw processing pipeline forego the whole map shizzl and basically just spew out `result.Row`s (which then can be formatted on the stream handler). The only limitation I see there is the column width, but I can live with using a conservative one-fits-all value TBH.

> I foresee some effort regarding results streaming. We for sure need to extend the data structure to support it meaningfully, as in:
> 
> * stream a bulk of rows per response
Some form of bulking for sure, but I think that comes naturally (we could just send the bulk of rows from each block over a channel and then print them bulk by bulk - the blocks have to be handled sequentially anyway to ensure output in the correct order).

> * stream increments of the summary?
I see two options here: Either we build up the summary on the side and then spew it out at the end _or_ we just skip the summary for raw queries entirely. Let's face it, the raw query is focused on the flows, not the summary (you can obtain that with a simple `list` or a trivial query if you need it).

**Comment by fako1024** [comment]
Some first attempts look promising in general. I was able to split up local queries relatively easily and extend the `goQuery` `Run()` to a loop with multiple runs (which in the non-raw case of course becomes a single loop iteration). There's a few things to be done of course:

- The `TablePrinter` needs a lot of love because the interface makes the assumption that there is just one `Result` (this goes beyond the pure question of table widths etc). I need to "break" that concept to allow "streaming" (in a way that preserves the current behavior, ideally without having to duplicate stuff). From a quick glance it's possible, it's just a lot of grind work I guess...
- When looking at the `TablePrinter` I saw that there's the question of host stati. Right now they are being printed in the finalization. However, when running multiple (chunk) queries for a raw query there's no guarantee that all "sub"-queries have the same host status (as in: for the first chunk one host could be down, for the second a different one, and so on). This will make the results a bit unpredictable (because due to streaming there is no way to "take back" what has already been printed). Since it's an edge case we can probably live with that and just assume that if the first query works subsequent ones will, too, but still it's noteworthy.
- If a raw query is run on multiple interfaces the whole thing is _really_ complicated. There's no guarantee that all interfaces cover the exact same time span / days (technically not even the same granularity), so you could easily end up with duplicate / missing data if just determining _one_ date range / next timestamp to query in pagination. This has to be remedied by looking at the available time ranges first and then determining the minimal common set per chunk (potentially skipping interfaces on that chunk if no data is available).
